warehouse/forklift/legacy.py

```diff
@@ -639,6 +631,17 @@ def file_upload(request):
                           func.normalize_pep426_name(form.name.data)).one()
         )
     except NoResultFound:
+        # Before we create the project, we're going to check our blacklist to
+        # see if this project is even allowed to be registered. If it is not,
+        # then we're going to deny the request to create this project.
+        if request.db.query(exists().where(
+                BlacklistedProject.name ==
+                func.normalize_pep426_name(form.name.data))).scalar():
+            raise _exc_with_message(
+                HTTPBadRequest,
+                "The name {!r} is not allowed.".format(form.name.data),
```

Comment:
Hi @dstufft - glad you started something!

Would suggest the message to be a little more clear?

    For concerns of security (namesquatting and typosquatting), '{!r}' is not an allowed name for a project. Please use a different name.

Reply:
I added that question to https://github.com/pypa/warehouse/issues/2401#issuecomment-330107332 (as I think it's related to the general UX of how we communicate invalid name declarations to end users)